systems/subscriber/sim-manager/pkg/db/sim_repo.go

```diff
@@ -124,3 +125,21 @@ func (s *simRepo) Delete(simID uuid.UUID, nestedFunc func(uuid.UUID, *gorm.DB) e
 
 	return err
 }
+func (s *simRepo) GetSimCounts() (simsCount int64, activeCount int64, deactiveCount int64, err error) {
+	var deactive int64
+
+	result := s.Db.GetGormDb().Model(&Sim{}).Count(&simsCount)
+	if result.Error != nil {
+		err = result.Error
+		return
+	}
+
+	result = s.Db.GetGormDb().Model(&Sim{}).Where("status =?", 2).Count(&activeCount)
+
+	if result.Error != nil {
+		err = result.Error
+		return
+	}
+	deactive = simsCount - activeCount
```

Comment:
Do we need a state for sim allocation which is for activation ? I am not sure.

Reply:
Here, we retrieve the total number of SIMs and subtract the number of activated SIMs to obtain the count of deactivated SIMs. The fields `ActivationsCount` and `DeactivationsCount`, which are for a single SIM, cannot be used in this case.

Reply:
I mant to say if you add more states in that case this will not be right ```deactive = simsCount - activeCount```

Reply:
Yes, you're correct. I will either separate the db calls or include a switch case to verify the state and return the appropriate value.